addons/sale_stock/models/stock_warehouse.py

```diff
@@ -0,0 +1,11 @@
+from odoo import models
+
+class Warehouse(models.Model):
+    _inherit = "stock.warehouse"
+
+    def _get_routes_values(self):
+        routes = super()._get_routes_values()
+        if routes.get('crossdock_route_id'):
+            routes['crossdock_route_id']['route_update_values']['sale_selectable'] = True
```

Comment:
I would leave that out of the `route_update_values`, I'm not sure it would make sense to reset it each time `_get_route_values()` is called.